src/stream/src/executor/approx_percentile/global_state.rs

```diff
@@ -0,0 +1,340 @@
+        self.row_count = row_count;
+        tracing::debug!(?row_count, "recovered row_count");
+
+        // Refill cache
+        self.refill_cache().await?;
+
+        // Update the last output downstream
+        let last_output = if row_count_state.is_none() {
+            None
+        } else {
+            Some(self.cache.get_output(row_count, self.quantile, self.base))
+        };
+        tracing::debug!(?last_output, "recovered last_output");
+        self.last_output = last_output;
+        Ok(())
+    }
+
+    async fn refill_cache(&mut self) -> StreamExecutorResult<()> {
+        let neg_bounds: (Bound<OwnedRow>, Bound<OwnedRow>) = (
+            Bound::Unbounded,
+            Bound::Excluded([Datum::from(ScalarImpl::Int16(0))].to_owned_row()),
+        );
+        let non_neg_bounds: (Bound<OwnedRow>, Bound<OwnedRow>) = (
+            Bound::Included([Datum::from(ScalarImpl::Int16(0))].to_owned_row()),
+            Bound::Unbounded,
+        );
+        #[for_await]
+        for keyed_row in self
+            .bucket_state_table
+            .rev_iter_with_prefix(&[Datum::None; 0], &neg_bounds, PrefetchOptions::default())
+            .await?
+            .chain(
+                self.bucket_state_table
+                    .iter_with_prefix(
+                        &[Datum::None; 0],
+                        &non_neg_bounds,
+                        PrefetchOptions::default(),
+                    )
+                    .await?,
+            )
+        {
+            let row = keyed_row?.into_owned_row();
+            let sign = row.datum_at(0).unwrap().into_int16();
+            let bucket_id = row.datum_at(1).unwrap().into_int32();
+            let count = row.datum_at(2).unwrap().into_int64();
+            match sign {
+                -1 => {
+                    self.cache.neg_buckets.insert(bucket_id, count as i64);
+                }
+                0 => {
+                    self.cache.zeros = count as i64;
+                }
+                1 => {
+                    self.cache.pos_buckets.insert(bucket_id, count as i64);
+                }
+                _ => {
+                    bail!("Invalid sign: {}", sign);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn get_row_count_state(&self) -> StreamExecutorResult<Option<OwnedRow>> {
+        self.count_state_table.get_row(&[Datum::None; 0]).await
+    }
+
+    fn decode_row_count(row_count_state: &Option<OwnedRow>) -> StreamExecutorResult<i64> {
+        if let Some(row) = row_count_state.as_ref() {
+            let Some(datum) = row.datum_at(0) else {
+                bail!("Invalid row count state: {:?}", row)
+            };
+            Ok(datum.into_int64())
+        } else {
+            Ok(0)
+        }
+    }
+}
+
+// Update
+impl<S: StateStore> GlobalApproxPercentileState<S> {
+    pub fn apply_chunk(&mut self, chunk: StreamChunk) -> StreamExecutorResult<()> {
+        for (_op, row) in chunk.rows() {
```

Comment:
Why ignoring `op` here?

Reply:
Because from the upstream, the input is always an `INSERT`. We only care about the `delta` column of the row, which will update our counts accordingly. (e.g. -100, +10, +5, etc...) 

I will add a comment on it.

It's a special way of processing stream chunks. The behaviour does not change from previous approx_percentile implementation.

Reply:
IC. May add a `debug_assert` here to ensure the correctness.

Reply:
Fixed.